[BUG] Line Break in Document title causes Workflow with E-Mail sending to break

### Description

When trying to send documents via E-Mail (either via Workflow or with the "send" functionality) sending breaks in the background without user notification when the title of the document containts a line break. 

First of all: I am unsure where this line break comes from as I cannot see it anywhere else within the document properties.

Second: the workflow continues and sets attributes even when mailing (the primary action) fails. this is an issue within itself. 

### Steps to reproduce

I am unsure on how to even produce documents with \n in their titles. 
The affected document was imported via E-Mail digest - the original E-mail does NOT contain \n in its title nor do I see a visual line break anywhere in webmail, outlook or paperless itself. 

<img width="552" height="107" alt="Image" src="https://github.com/user-attachments/assets/7402b848-e4cb-404c-8a9f-b2700293d104" />

Here's the supposed line break but it does not show there. 

The Mail Header itself also does not show any signs of any line breaks:

<img width="388" height="86" alt="Image" src="https://github.com/user-attachments/assets/70b4c4b3-925c-456d-930b-8d7c3ed73c74" />

The only hint that something might be off is here:

<img width="563" height="79" alt="Image" src="https://github.com/user-attachments/assets/7cc4410d-c203-4849-b94b-0e0f5cfecab0" />

The Title itself is:
"SYSTEAM GmbH Rechnung xXXXX/XXXXX, Ihr Auftragskennzeichen: Basket XXXXX/XXXX, Order Date 08.06.2026" (this is a copy & paste from the title field, no line break here)

So the filename should be ", Ihr Auftragskennzeichen- Basket" since : gets replaced by - but not "or ", Ihr Auftragskennzeichen**--** Basket"


I suggest the following changes/fixes:
- Error handler for Mail sending to stop Workflows / notify the user
- String Cleanup before sending
- String Cleanup for the title field itself

### Webserver logs

```bash
[2026-07-29 12:06:53,506] [INFO] [paperless.handlers] Applying WorkflowAction 2 from Workflow: EINGANG Rechnungen an XYZ

[2026-07-29 12:06:53,511] [DEBUG] [paperless.templating] Parsing Workflow Jinja template: Eingangsrechnung - {{doc_title}}

[2026-07-29 12:06:53,511] [DEBUG] [paperless.templating] Parsing Workflow Jinja template: Dokumententitel: {{doc_title}}

Dokumententyp: {{document_type}}

Korrespondent: {{correspondent}}:

Dokumenten-Datum: {{created}}


Zuordnung:

{{filename}}

[2026-07-29 12:06:53,759] [ERROR] [paperless.workflows.actions] Error occurred sending notification email: Header values can't contain newlines (got 'Eingangsrechnung - SYSTEAM GmbH Rechnung 60237352/132282, Ihr Auftragskennzeichen:\n Basket 26049646/001, Order Date 08.06.2026' for header 'Subject')

Traceback (most recent call last):

  File "/usr/src/paperless/src/documents/workflows/actions.py", line 165, in execute_email_action

    n_messages = send_email(

        subject=subject,

    ...<2 lines>...

        attachments=attachments,

    )

  File "/usr/src/paperless/src/documents/mail.py", line 68, in send_email

    return email.send()

           ~~~~~~~~~~^^

  File "/usr/local/lib/python3.13/dist-packages/django/core/mail/message.py", line 307, in send

    return self.get_connection(fail_silently).send_messages([self])

           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^

  File "/usr/local/lib/python3.13/dist-packages/django/core/mail/backends/smtp.py", line 136, in send_messages

    sent = self._send(message)

  File "/usr/local/lib/python3.13/dist-packages/django/core/mail/backends/smtp.py", line 153, in _send

    message = email_message.message()

  File "/usr/local/lib/python3.13/dist-packages/django/core/mail/message.py", line 270, in message

    msg["Subject"] = self.subject

    ~~~^^^^^^^^^^^

  File "/usr/local/lib/python3.13/dist-packages/django/core/mail/message.py", line 190, in __setitem__

    name, val = forbid_multi_line_headers(name, val, self.encoding)

                ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.13/dist-packages/django/core/mail/message.py", line 61, in forbid_multi_line_headers

    raise BadHeaderError(

        "Header values can't contain newlines (got %r for header %r)" % (val, name)

    )
```

### Browser logs

```bash

```

### Services logs

```bash

```

### Paperless-ngx version

v2.20.15

### Host OS

home assistant add-in

### Installation method

Other (please describe above)

### System status

```json

```

### Browser

_No response_

### Configuration changes

_No response_

### Please confirm the following

- [x] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [x] This issue is not about the OCR or archive creation of a specific file(s). Otherwise, please see above regarding OCR tools.
- [x] I have already searched for relevant existing issues and discussions before opening this report.
- [x] I have updated the title field above with a concise description.

--- a/src/documents/mail.py
+++ b/src/documents/mail.py
@@ -1,6 +1,5 @@
 from __future__ import annotations
 
-import re
 from dataclasses import dataclass
 from email import message_from_bytes
 from pathlib import Path
@@ -37,7 +36,8 @@ def send_email(
 
     TODO: re-evaluate this pending https://code.djangoproject.com/ticket/35581 / https://github.com/django/django/pull/18966
     """
-    subject = re.sub(r"[ \t]*[\r\n]+[ \t]*", " ", subject)
+    if "\r" in subject or "\n" in subject:
+        subject = " ".join(line.strip(" \t") for line in subject.splitlines())
 
     email = EmailMessage(
         subject=subject,

--- a/src/documents/mail.py
+++ b/src/documents/mail.py
@@ -1,5 +1,6 @@
 from __future__ import annotations
 
+import re
 from dataclasses import dataclass
 from email import message_from_bytes
 from pathlib import Path
@@ -36,6 +37,8 @@ def send_email(
 
     TODO: re-evaluate this pending https://code.djangoproject.com/ticket/35581 / https://github.com/django/django/pull/18966
     """
+    subject = re.sub(r"[ \t]*[\r\n]+[ \t]*", " ", subject)
+
     email = EmailMessage(
         subject=subject,
         body=body,

--- a/src/documents/tests/test_api_email.py
+++ b/src/documents/tests/test_api_email.py
@@ -75,7 +75,7 @@ class TestEmail(DirectoriesMixin, SampleDirMixin, APITestCase):
                 {
                     "documents": [self.doc1.pk, self.doc2.pk],
                     "addresses": "hello@paperless-ngx.com,test@example.com",
-                    "subject": "Bulk email test",
+                    "subject": "Bulk email\n test",
                     "message": "Here are your documents",
                 },
             ),

